### DISCUSSION TITLE
Make root mail feature optional, e.g. if mail provider supports plus addressing "aws+department@example.com"

### DISCUSSION THREAD

**Discussion by bracki** [superwerker/superwerker#283] [Ideas]
# Proposed change

- Add a CFN parameter `IncludeRootMail` that defaults to `true`
- Provide an optional  CFN parameter `RootAccountMailPattern` (or similar) that you would set to "something@example.com"
- If `RootAccountMailPattern` is set, disable `IncludeRootMail`.
- Add a validation hook/mechanism that rejects account creation, if the mail does not match `RootAccountMailPattern`. The pattern would be `/something(\+[A-Za-z0-9]+)?@example.com/`

# Why?

The root mail feature makes an automated/scripted setup of superwerker cumbersome. Enforcing an email pattern for newly created accounts could be sufficient.

# Why not?

Maybe other features directly depend on the root mail feature.

**Comment by s0enke** [comment]
Thanks for your suggestion!

The `RootMail` feature is currently mandatory because:

 - the ControlTower stack depends upon it, since RootMail provides the two email addresses (audit, log archive) necessary for Control Tower setup.
 - (future) automations, such as automated AWS account creation for workloads.

I put a clarification into the ADR here: https://github.com/superwerker/superwerker/commit/091872af7720b9b35d6b56b44b2303253575e12c

It's still possible to create AWS accounts outside the email scheme provided by the RootMail feature.
